utils/java-dynamic-memory-opts

```diff
@@ -14,11 +14,16 @@ if [ -z "$MEM_JAVA_PERCENT" ]; then
     fi
 fi
 
-# MEM_TOTAL_KB can be set from the outside, this is important for container-based infrastructure:
-# /proc/meminfo would return the host/node capacity and not the container's limit
-if [ -z "$MEM_TOTAL_KB" ]; then
-    MEM_TOTAL_KB=$(cat /proc/meminfo | grep MemTotal | awk '{print $2}')
-fi
-MEM_JAVA_KB=$(($MEM_TOTAL_KB * $MEM_JAVA_PERCENT / 100))
+JVERSION=$(java -version 2>&1 | head -n 1 | awk -F '"' '{print $2}')
+if [ $JVERSION == 1.8* ]; then
```

Comment:
I don't think you can use pattern matching with plain `[`, but bash's `[[` is needed:
```
$ jversion=1.8.0_181
$ [ $jversion == 1.8* ]; echo $?
1
$ [[ $jversion == 1.8* ]]; echo $?
0
```

but if the file is supposed to be for `sh`, then something else might be needed.

Reply:
I'm not sure tbh...
There didn't seem to be much enthusiasm to accept this PR so I've moved to using non stups base jdk images. If you'd like you can take it over. If not I'll prob close off this PR.

Reply:
Our team is pretty happy with the debian-slim based openjdk image.

Reply:
There is also AdoptOpenJDK https://hub.docker.com/r/adoptopenjdk/openjdk11 which provides ubuntu-based images.